### PR TITLE
Populate PUB_ENVIRONMENT for initial pub get and correctly detect APPVEYOR as bot

### LINKED ISSUE #9668: Include "FLUTTER_INSTALL" in PUB_ENVIRONMENT var when running `pub upgrade` on flutter_tools during install
We're getting a LOT of hits for this tool in our pub telemetry. It gets run A LOT so it really skews our metrics.

### PR DESCRIPTION
**Pull request by goderbauer**
fixes #9668

### CODE REVIEW

File: bin/flutter [hunk 1]
```diff
@@ -81,7 +81,11 @@ fi
     "$FLUTTER_ROOT/bin/internal/update_dart_sdk.sh"
 
     echo Building flutter tool...
-    (cd "$FLUTTER_TOOLS_DIR"; "$PUB" upgrade --verbosity=error --no-packages-dir)
+    pub_env="flutter_install:$PUB_ENVIRONMENT"
```
Comment (Hixie) [inline bin/flutter]: looks like all the other variables are uppercase, we may want to be consistent.
not that we have a bash style guide other than "not use bash". :-)

File: bin/flutter.bat [hunk 2]
```diff
@@ -90,7 +90,23 @@ GOTO :after_subroutine
     ECHO: > "%cache_dir%\.dartignore"
     ECHO Updating flutter tool...
     PUSHD "%flutter_tools_dir%"
+
+    REM Makes changes to PUB_ENVIRONMENT only visiable to commands within SETLOCAL/ENDLOCAL
```
Comment (Hixie) [inline bin/flutter.bat]: visible

**Comment by Hixie** [comment]
LGTM

**Review by kevmoo — CHANGES_REQUESTED** [review]
Fix order. Mark this as fixing https://github.com/flutter/flutter/issues/9668

**Review by kevmoo — APPROVED** [review]

**Pull request merged**